docs/gc.md

```diff
@@ -131,32 +124,19 @@ Object ages are tracked for each region with a maximum of 24 possible generation
 
 ![The diagram is explained in the surrounding text](./cr/balanced_regions.png "Region structure and characteristics found in the object heap")
 
-The *eden* space is a set of regions of age 0, which contain the newest objects allocated. When the *eden* space is full, a partial GC cycle runs to reclaim space, typically by using a *copy forward* operation. In specific cases, a *mark-compact* operation might be used, for example, when there are not enough free survivor regions available. The partial GC cycle is a *stop-the-world* (STW) operation, after which new regions are allocated to create a new *eden* space. A partial cycle always includes the *eden* space, but might include older regions. Objects from collectible regions of age N are moved into another region of the same age N or to an empty region that is assigned an age of N. Then, the ages of all regions across the heap are incremented by 1, except for the maximum age 24 regions. Regions of age 24 are included in partial GC collection sets in order to defragment them.
+The *eden* space is a set of regions of age 0, which contain the newest objects allocated. When the *eden* space is full, a partial GC cycle runs to reclaim space, typically by using a *copy forward* operation. In specific cases, *mark* and *compact* operations might be used, for example, when there are not enough free survivor regions available. The partial GC cycle is a STW operation, after which new regions are allocated to create a new *eden* space. A partial cycle always includes the *eden* space, but might include older regions. Objects from collectible regions of age N are moved into another region of the same age N or to an empty region that is assigned an age of N. Then, the ages of all regions across the heap are incremented by 1, except for the maximum age 24 regions. Regions of age 24 are included in partial GC collection sets in order to defragment them.
```

Comment:
This part seems redundant: ', after which new regions are allocated to create a new *eden* space."

One thing to realize is that eden space is not created just after a PGC (as it is the case with allocate in scavenge&nursery). Eden  is just a collection of regions (that could be scattered all over heap) that have recently allocated objects. Those regions are picked from the pool of all empty regions as allocation progresses. PGC just reclaims memory and create empty regions that are put to that pool.

PGC is therefore triggered when eden size threshold (region count) is reached, not when specific predetermined area (what one would assume is eden space) of heap is full, because eden space has not been predetermined. 

Similar thing could be said for survivor regions. They don't really exist - they are not predetermined. They are just consumed from the same global pool of free regions, but just during PGC time. As said in next paragraph, PGC at the end of cycle will return some (most, but not all) of those consumed regions back to the pool.

This is one important advantage of Balanced over Gencon (generally, region based GCs over area based), and could be emphasis: survivor and allocate/eden areas are not  predetermined and share free memory from the global pool of free regions, what in turn reduces probability of (too) late concurrent mark cycles and aborted partial GCs.

Reply:
Hopefully, this is enough to fix:

"The *eden* space is a set of regions of age 0, which contain the newest objects allocated. The size of the *eden* space is determined by the number of regions that it contains. When the region count for the *eden* space reaches a predetermined threshold, a partial GC cycle runs to reduce the number of regions, typically by using a *copy forward* operation. Empty regions can then be assigned to the *eden* space from the pool."

Reply:
runs to reduce the number of _used (or occupied)_ regions